--- a/src/QMIDevice.h
+++ b/src/QMIDevice.h
@@ -322,7 +322,7 @@ int QuecQMIWDASetDataFormat( sGobiUSBNet * pDev, int qmap_mode, int *rx_urb_si
 
 // Print Hex data, for debug purposes
 void QuecPrintHex(
-   void *         pBuffer,
+   const void *   pBuffer,
    u16            bufSize );
 
 // Sets mDownReason and turns carrier off
--- a/src/QMIDevice.c
+++ b/src/QMIDevice.c
@@ -330,7 +330,7 @@ int QuecGetClientID(
 /*===========================================================================
 ===========================================================================*/
 void QuecPrintHex(
-   void *      pBuffer,
+   const void * pBuffer,
    u16         bufSize )
 {
    char * pPrintBuf;
